### DISCUSSION TITLE
wifi disconnect caused by GPIO assignment

### DISCUSSION THREAD

**Discussion by jpadie** [arendst/Tasmota#19143] [Support request]
assigning GPIO2 to MODBR_TX stops wifi from connecting on ESP32 using web installed 13.0.0 tasmota32. 

see log below (redacted for repetition).  I have validated this behaviour across 24 hours.  

any thoughts?  I don't think that GPIO2 is used for any wifi magic under the hood.  

`
12:57:04.818 -> 00:00:15.063 CMD: gpio
12:57:04.818 -> 00:00:15.064 SRC: Serial
12:57:04.818 -> 00:00:15.065 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 0, Pld -99, Data ''
12:57:04.818 -> 00:00:15.074 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"9408":"ModBr Tx"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"9440":"ModBr Rx"},"GPIO16":{"0":"None"},"GPIO17":{"9952":"ModBr Tx Ena"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO39":{"0":"None"}}
12:57:05.677 -> 00:00:15.934 WIF: Checking connection...
12:57:05.677 -> 00:00:15.936 WIF: Connect failed with AP timeout
12:57:06.040 -> 00:00:16.291 WIF: Connecting to AP1 Gaia-Rabastens-Free in mode 11n as poolManager-2212...
12:57:06.934 -> 00:00:17.196 WIF: Checking connection...
12:57:06.934 -> 00:00:17.198 WIF: Attempting connection...
12:57:07.982 -> 00:00:18.214 WIF: Checking connection...
12:57:07.982 -> 00:00:18.216 WIF: Attempting connection...
12:57:09.775 -> 00:00:20.031 WIF: Checking connection...
12:57:09.775 -> 00:00:20.033 WIF: Connect failed as AP cannot be reached
12:57:11.138 -> 00:00:21.168 WIF: Checking connection...
12:57:11.138 -> 00:00:21.169 WIF: Attempting connection...
12:57:11.497 -> 00:00:21.524 WIF: Connecting to AP1 Gaia-Rabastens-Free in mode 11n as poolManager-2212...
//NO WIFI CONNECTION
//REMOVE GPIO ASSIGNMENT
12:57:12.747 -> 00:00:22.541 CMD: gpio2 0
12:57:12.747 -> 00:00:22.542 SRC: Serial
12:57:12.747 -> 00:00:22.543 CMD: Grp 0, Cmd 'GPIO', Idx 2, Len 1, Pld 0, Data '0'
12:57:12.747 -> 00:00:22.552 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"9440":"ModBr Rx"},"GPIO16":{"0":"None"},"GPIO17":{"9952":"ModBr Tx Ena"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO39":{"0":"None"}}
12:57:12.996 -> 00:00:23.244 WIF: Checking connection...
12:57:12.996 -> 00:00:23.246 WIF: Attempting connection...
12:57:14.284 -> 00:00:24.349 CFG: Saved, Count 49, Bytes 4096
12:57:14.390 -> 00:00:24.466 WIF: Checking connection...
12:57:14.390 -> 00:00:24.467 WIF: Attempting connection...
12:57:15.809 -> ets Jun  8 2016 00:22:57
12:57:15.809 -> 
12:57:15.809 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
12:57:15.809 -> configsip: 0, SPIWP:0xee
12:57:15.809 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
12:57:15.809 -> mode:DIO, clock div:2
12:57:15.809 -> load:0x3fff0030,len:184
12:57:15.809 -> load:0x40078000,len:12732
12:57:15.809 -> ho 0 tail 12 room 4
12:57:15.809 -> load:0x40080400,len:2908
12:57:15.809 -> entry 0x400805c4
12:57:15.902 -> 
12:57:15.902 -> 00:00:00.003 HDW: ESP32-D0WDQ6 
12:57:15.902 -> 00:00:00.026 UFS: FlashFS mounted with 304 kB free
12:57:15.902 -> 00:00:00.040 CFG: Loaded from File, Count 49
12:57:15.959 -> 00:00:00.050 QPC: Count 1
12:57:15.959 -> 00:00:00.052 CFG: CR 433/699, Busy 0
12:57:15.959 -> 00:00:00.066 ROT: Mode 1
12:57:15.959 -> 00:00:00.075 BRY: GC from 3665 to 3465 bytes, objects freed 4/20 (in 0 ms) - slots from 35/61 to 27/61
12:57:16.016 -> 00:00:00.137 CFG: No '*.autoconf' file found
12:57:16.016 -> 00:00:00.143 BRY: GC from 5801 to 4406 bytes, objects freed 6/40 (in 1 ms) - slots from 63/91 to 41/91
12:57:16.016 -> 00:00:00.144 BRY: Berry initialized, RAM used=4406 bytes
12:57:16.060 -> 00:00:00.164 BRY: No 'preinit.be'
12:57:16.060 -> 00:00:00.168 TFS: File '.drvset003' not found
12:57:16.060 -> 00:00:00.168 CFG: Energy use defaults as file system not ready or file not found
12:57:16.060 -> 00:00:00.172 SRC: Restart
12:57:16.060 -> 00:00:00.176 Project tasmota - Tasmota Version 13.0.0(tasmota32)-2_0_10(2023-06-26T10:54:38)
12:57:16.060 -> 00:00:00.188 ETH: No ETH MDC and/or ETH MDIO GPIO defined
12:57:16.105 -> 00:00:00.237 BRY: No 'autoexec.be'
12:57:16.386 -> 00:00:00.492 WIF: Checking connection...
12:57:16.386 -> 00:00:00.493 WIF: Attempting connection...
12:57:16.819 -> 00:00:00.944 WIF: Connecting to AP1 Gaia-Rabastens-Free in mode 11n as poolManager-2212...
12:57:17.641 -> 00:00:01.109 WIF: Checking connection...
12:57:17.641 -> 00:00:01.110 WIF: Attempting connection...
12:57:18.642 -> 00:00:02.709 WIF: Checking connection...
12:57:18.642 -> 00:00:02.710 WIF: Attempting connection...
12:57:19.646 -> 00:00:03.726 WIF: Checking connection...
12:57:19.646 -> 00:00:03.727 WIF: Attempting connection...
12:57:20.643 -> 00:00:04.694 WIF: Checking connection...
12:57:20.643 -> 00:00:04.695 WIF: Attempting connection...
12:57:21.250 -> 00:00:05.303 WIF: IPv4 192.168.0.30, mask 255.255.255.0, gateway 192.168.0.1
12:57:21.649 -> 00:00:05.710 WIF: Checking connection...
12:57:21.649 -> 00:00:05.712 WIF: Connected
12:57:21.900 -> 00:00:05.975 HTP: Web server active on poolManager-2212 with IP address 192.168.0.30

//NORMAL CONNECTION

12:57:21.900 -> 00:00:05.980 NTP: Sync time...
12:57:21.970 -> 00:00:06.031 WIF: DNS resolved '2.pool.ntp.org' (129.250.35.250) in 51 ms
12:57:22.401 -> 00:00:06.457 RTC: UTC 2023-07-18T10:57:22, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
12:57:22.401 -> 11:57:22.000 RTC: Synced by NTP
12:57:22.689 -> 11:57:22.305 WIF: IPv6 Local fe80::eb8:15ff:feec:88a4
12:57:22.800 -> 11:57:22.407 CFG: Saved, Count 50, Bytes 4096
12:57:22.906 -> 11:57:22.530 QPC: Reset
12:57:23.119 -> 11:57:22.180 MQT: Attempting connection...
//REDACTED

//REASSIGN GPIO
12:57:47.965 -> 11:57:47.045 CMD: gpio2 9408
12:57:47.965 -> 11:57:47.048 SRC: Serial
12:57:47.965 -> 11:57:47.049 CMD: Grp 0, Cmd 'GPIO', Idx 2, Len 4, Pld 9408, Data '9408'
12:57:48.001 -> 11:57:47.065 MQT: rab/poolManager/stat/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"9408":"ModBr Tx"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"9440":"ModBr Rx"},"GPIO16":{"0":"None"},"GPIO17":{"9952":"ModBr Tx Ena"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO39":{"0":"None"}}
12:57:49.175 -> 11:57:48.229 CFG: Saved, Count 52, Bytes 4096
12:57:49.882 -> 11:57:48.947 APP: Restarting
12:57:50.318 -> ets Jun  8 2016 00:22:57
12:57:50.318 -> 
12:57:50.318 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
12:57:50.318 -> configsip: 0, SPIWP:0xee
12:57:50.318 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
12:57:50.356 -> mode:DIO, clock div:2
12:57:50.356 -> load:0x3fff0030,len:184
12:57:50.356 -> load:0x40078000,len:12732
12:57:50.356 -> ho 0 tail 12 room 4
12:57:50.356 -> load:0x40080400,len:2908
12:57:50.356 -> entry 0x400805c4
12:57:50.406 -> 
12:57:50.406 -> 00:00:00.002 HDW: ESP32-D0WDQ6 
12:57:50.462 -> 00:00:00.045 UFS: FlashFS mounted with 304 kB free
12:57:50.462 -> 00:00:00.073 CFG: Loaded from File, Count 52
12:57:50.519 -> 00:00:00.083 QPC: Count 1
12:57:50.519 -> 00:00:00.085 CFG: CR 433/699, Busy 0
12:57:50.519 -> 00:00:00.100 ROT: Mode 1
12:57:50.519 -> 00:00:00.108 BRY: GC from 3665 to 3465 bytes, objects freed 4/20 (in 1 ms) - slots from 35/61 to 27/61
12:57:50.602 -> 00:00:00.216 CFG: No '*.autoconf' file found
12:57:50.638 -> 00:00:00.222 BRY: GC from 5801 to 4406 bytes, objects freed 6/40 (in 1 ms) - slots from 63/91 to 41/91
12:57:50.638 -> 00:00:00.222 BRY: Berry initialized, RAM used=4406 bytes
12:57:50.638 -> 00:00:00.246 BRY: No 'preinit.be'
12:57:50.674 -> 00:00:00.252 TFS: File '.drvset003' not found
12:57:50.674 -> 00:00:00.252 CFG: Energy use defaults as file system not ready or file not found
12:57:50.674 -> 00:00:00.256 MBS: MBR SW ser init at 19200 baud
12:57:50.674 -> 00:00:00.258 SRC: Restart
12:57:50.674 -> 00:00:00.273 Project tasmota - Tasmota Version 13.0.0(tasmota32)-2_0_10(2023-06-26T10:54:38)
12:57:50.674 -> 00:00:00.275 ETH: No ETH MDC and/or ETH MDIO GPIO defined
12:57:50.783 -> 00:00:00.372 BRY: No 'autoexec.be'
12:57:50.890 -> 00:00:00.493 WIF: Checking connection...
12:57:50.890 -> 00:00:00.494 WIF: Attempting connection...
12:57:51.366 -> 00:00:00.958 WIF: Connecting to AP1 Gaia-Rabastens-Free Channel 1 BSSId 6C:61:F4:00:2D:26 in mode 11n as poolManager-2212...
12:57:52.185 -> 00:00:01.040 WIF: Checking connection...
12:57:52.185 -> 00:00:01.041 WIF: Attempting connection...
12:57:53.144 -> 00:00:02.652 WIF: Checking connection...
12:57:53.144 -> 00:00:02.653 WIF: Attempting connection...
12:57:54.434 -> 00:00:03.552 WIF: Checking connection...
12:57:54.434 -> 00:00:03.553 WIF: Attempting connection...
12:57:55.381 -> 00:00:04.896 WIF: Checking connection...
12:57:55.381 -> 00:00:04.897 WIF: Attempting connection...
12:57:56.378 -> 00:00:05.782 WIF: Checking connection...
12:57:56.378 -> 00:00:05.783 WIF: Attempting connection...
12:57:57.413 -> 00:00:06.922 WIF: Checking connection...
12:57:57.413 -> 00:00:06.923 WIF: Attempting connection...
12:57:57.413 -> 00:00:06.934 QPC: Reset
12:57:58.448 -> 00:00:07.939 WIF: Checking connection...
`

**Comment by SimonFili** [comment]
GPIO2 is often used for bootstrapping or LED. Did you try with another GPIO?

**Comment by Jason2866** [comment]
Using GPIO 2 is maybe a problem in your opened #19137 too?

GPIO 2 (must be floating or LOW during boot)

A nice write up about https://randomnerdtutorials.com/esp32-pinout-reference-gpios/

**Comment by jpadie** [comment]
Remember that all that is changing is the pin allocation.  The physical hardware on the pin remains there.  Tasmota sets the pin to output mode in its set up routine.  This is after all the bootstrapping.  Wifi is usually affected by interacting with adc2. I don't think this is an issue here.  On 18 Jul 2023 17:42, Simon Filiatrault ***@***.***> wrote:
GPIO2 is often used for bootstrapping or LED. Did you try with another GPIO?

**Comment by Jason2866** [comment]
I wouldn't do assumptions. Rewire to an unrestricted GPIO will show if the problem is caused from or not. We can't simulate your setup nor the issue.
The board is badly designed. ESP32 has so many unrestricted GPIOS.
GPIO 2 IS doing unknown stuff under the hood. We don't know anything here.
